lcdb_test_data/Snakefile

```diff
@@ -1,19 +1,31 @@
 import os
 from textwrap import dedent
 
+# required to avoid near-simultaneous timestamps that confuse snakemake
 shell.prefix('sleep 2;')
 
+
 rule all:
-    input: expand('samples/sample{n}/sample{n}.bam', n=range(1, 5))
+    input:
+        expand(
+            'samples/sample{n}/sample{n}.{paired}.bam',
+            n=range(1, 5), paired=['paired', 'single']
+        )
+
 
+# ----------------------------------------------------------------------------
+# Create a BED file that will be used to subset GTF and FASTA files
 rule limits:
     output: temp('LIMIT.bed')
     shell:
         'echo "2L	0	80000	2L" > {output}'
 
+
+# ----------------------------------------------------------------------------
+# Download FlyBase GTF, so
 rule prep_gtf:
     input: rules.limits.output
```

Comment:
yep, and if the `limits` rule changes its output, `prep_gtf` doesn't have to be changed.

I think it makes rule dependencies more explicit to humans reading the snakefile. Only time it gets annoying is when you're debugging rule and the input filename pattern is not right there in the rule. Running snakemake with `-n` usually solves that though.